tests/integration_tests/_utils/run_pulsar_cluster

```diff
@@ -0,0 +1,151 @@
+#!/bin/bash
+
+# parameter 1: work directory
+# parameter 2: cluster_type,  mtls or oauth, otherwise use default configuration to start pulsar cluster
+
+set -eux
+
+echo "[$(date)] <<<<<< START pulsar cluster in $TEST_NAME case >>>>>>"
+workdir=$1
+cluster_type=$2
+
+cd $workdir
+
+DEFAULT_PULSAR_HOME="/usr/local/pulsar"
+# use PULSAR_HOME if it is set, otherwise use default pulsar home
+pulsar_dir=${PULSAR_HOME:-$DEFAULT_PULSAR_HOME}
+
+mtls_conf=$(
+	cat <<-EOF
+
+		authenticationEnabled=true
+		authenticationProviders=org.apache.pulsar.broker.authentication.AuthenticationProviderTls
+		brokerClientTlsEnabled=true
+		brokerClientTrustCertsFilePath=${workdir}/ca.cert.pem
+		brokerClientAuthenticationPlugin=org.apache.pulsar.client.impl.auth.AuthenticationTls
+		brokerClientAuthenticationParameters={"tlsCertFile":"${workdir}/broker_client.cert.pem","tlsKeyFile":"${workdir}/broker_client.key-pk8.pem"}
+		brokerServicePortTls=6651
+		webServicePortTls=8443
+		tlsTrustCertsFilePath=${workdir}/ca.cert.pem
+		tlsCertificateFilePath=${workdir}/server.cert.pem
+		tlsKeyFilePath=${workdir}/server.key-pk8.pem
+		tlsRequireTrustedClientCertOnConnect=true
+		tlsAllowInsecureConnection=false
+		tlsCertRefreshCheckDurationSec=300
+	EOF
+)
+
+normal_client_conf=$(
+	cat <<-EOF
+
+		webServiceUrl=http://localhost:8080/
+		brokerServiceUrl=pulsar://localhost:6650/
+	EOF
+)
+
+mtls_client_conf=$(
+	cat <<-EOF
+
+		webServiceUrl=https://localhost:8443/
+		brokerServiceUrl=pulsar+ssl://localhost:6651/
+		authPlugin=org.apache.pulsar.client.impl.auth.AuthenticationTls
+		authParams=tlsCertFile:${workdir}/broker_client.cert.pem,tlsKeyFile:${workdir}/broker_client.key-pk8.pem
+		tlsTrustCertsFilePath=${workdir}/ca.cert.pem
+	EOF
+)
+
+oauth_conf=$(
+	cat <<-EOF
+
+		authenticationEnabled=true
+		authenticationProviders=org.apache.pulsar.broker.authentication.AuthenticationProviderToken
+
+		brokerClientAuthenticationPlugin=org.apache.pulsar.client.impl.auth.oauth2.AuthenticationOAuth2
+		brokerClientAuthenticationParameters={"privateKey":"file://${workdir}/privateKey","audience":"https://dev-kt-aa9ne.us.auth0.com/api/v2/","issuerUrl":"https://dev-kt-aa9ne.us.auth0.com"}
```

Comment:
Will the auth0 dependency be replaced with an internal implementation in the future?

Reply:
yes, I will start an oauth2 server in integration tests in next PR